org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/async/StepMessageOut.scala

```diff
@@ -50,38 +59,57 @@ case class StepMessageOut(debugTarget: ScalaDebugTarget, thread: ScalaThread) ex
     thread.resumeFromScala(DebugEvent.CLIENT_REQUEST)
   }
 
+  private[async] def findThisType(stepEvent: StepEvent): String = {
+    val loc = stepEvent.location()
+    val declType = loc.declaringType().name()
+    Try {
```

Comment:
You could just use try-catch here, couldn't you? Also why do you just drop the throwable, how can it happen?